client/my-sites/plugins/main.jsx

```diff
@@ -164,11 +166,17 @@ export default React.createClass( {
 		}
 	},
 
-	getSelected() {
+	getSelected( isWpCom ) {
 		if ( ! this.state.plugins ) {
 			return [];
 		}
-		return this.state.plugins.filter( this.filterSelection.selected );
+
+		return this.state.plugins.filter( plugin => {
+			if ( plugin.selected ) {
+				return isWpCom ? plugin.wpcom : ! plugin.wpcom
+			}
+			return false;
+		} );
 	},
```

Comment:
We can get rid of the error checking here with destructuring and then simplify the function as a whole:

```js
function selectedAndMatchesWpCom( isWpCom ) {
	return plugin => plugin.selected && ( isWpCom ? plugin.wpcom : ! plugin.wpcom );
}

getSelected( isWpCom ) {
	const { plugins = [] } = this.state;

	return plugins.filter( selectedAndMatchesWpcom( isWpCom ) )
}
```

I'm not sure the best thing to name `selectedAndMatchesWpCom` (concerning the `isWpCom` part) mainly because I'm not sure what it's supposed to accomplish. You can probably think up a better name.